### PR TITLE
Made bulk mode configurable

### PR DESCRIPTION
**Pull request by psquickitjayant**
1. Adding 100 logs in array if the `isBulk` is set to true
2. Sends the logs after 30 seconds if logs are less than 100

### CODE REVIEW

**Comment by mostlyjason** [comment]
@indexzero possible to get this merged? We are trying to address this issue https://github.com/winstonjs/winston-loggly/issues/6. This should also help a lot with performance.

File: lib/loggly/common.js [hunk 1]
```diff
@@ -5,6 +5,13 @@
  * MIT LICENSE
  *
  */
+ 
+//
+// Variables for Bulk
+//
+var arr_size = 100;
```
Comment (indexzero) [inline lib/loggly/common.js]: We use `camelCase` so `arrSize`

File: lib/loggly/common.js [hunk 2]
```diff
@@ -5,6 +5,13 @@
...
+var arr_msg=[];
```
Comment (indexzero) [inline lib/loggly/common.js]: We use `camelCase` so `arrMsg`

File: lib/loggly/common.js [hunk 3]
```diff
@@ -5,6 +5,13 @@
...
+var timer_function = null;
```
Comment (indexzero) [inline lib/loggly/common.js]: We use `camelCase` so `timerFunction`

File: lib/loggly/common.js [hunk 4]
```diff
@@ -111,34 +119,77 @@ common.loggly = function () {
...
-    headers: headers || {}
+    headers: isBulk ? {} : headers || {}             // Set headers empty for bulk
   };
 
   if (auth) {
     requestOptions.headers['authorization'] = 'Basic ' + new Buffer(auth.username + ':' + auth.password).toString('base64');
   }
+  
+  function sendBulkLogs(){
+      if(arr_msg.length == 0){
```
Comment (indexzero) [inline lib/loggly/common.js]: Two space soft tabs.

**Comment by indexzero** [comment]
Please conform to the style of the existing code base.

**Comment by psquickitjayant** [comment]
@indexzero  thanks for the styling check. We have fixed the 2 soft tab spaces and camel case issues.

**Comment by indexzero** [comment]
@psquickitjayant still some small errors here. You can test them by running:

```
npm i -g common-style
common lib/
```

My apologies for not making this more transparent. I am going to update this repo now to use that as part of the CI/CD process.

**Comment by indexzero** [comment]
@psquickitjayant if you rebase now and run:

```
npm run pretest
```

You will see the errors relating to code style. Would love some help getting this repo fully setup on TravisCI since we need to encrypt the loggly credentials.

@mostlyjason would love some help from Loggly on this :smile_cat:

**Comment by psquickitjayant** [comment]
closing this pull request. I will send a new one with latest code changes

**Pull request closed without merging**